packages/flutter/test/material/expansion_panel_test.dart

```diff
@@ -402,4 +402,137 @@ void main() {
+    expect(find.text('B'), findsNothing);
+    expect(find.text('C'), findsOneWidget);
+    expect(find.text('D'), findsNothing);
+
+    await tester.tap(find.byKey(firstPanelKey));
+    await tester.pumpAndSettle();
+
+    // Now the first panel is open
+    expect(find.text('A'), findsNothing);
+    expect(find.text('B'), findsOneWidget);
+    expect(find.text('C'), findsOneWidget);
+    expect(find.text('D'), findsNothing);
+
+    await tester.tap(find.byKey(secondPanelKey));
+    await tester.pumpAndSettle();
+
+    // Now the second panel is open
+    expect(find.text('A'), findsOneWidget);
+    expect(find.text('B'), findsNothing);
+    expect(find.text('C'), findsNothing);
+    expect(find.text('D'), findsOneWidget);
+  });
+
+  testWidgets('Toggle simple panel list when clicking on header',  (WidgetTester tester) async {
+    final List<bool> listState = <bool>[false, false];
+
+    const Key firstPanelKey = Key('firstPanelKey');
+    const Key secondPanelKey = Key('secondPanelKey');
+
+    final List<ExpansionPanel> _demoItemsRadio = <ExpansionPanel>[
+      ExpansionPanel(
+        headerBuilder: (BuildContext context, bool isExpanded) {
+          return Text(isExpanded ? 'B' : 'A', key: firstPanelKey,);
+        },
+        body: const SizedBox(height: 100.0),
+        canTapOnHeader: true,
+        isExpanded: listState[0],
+      ),
+      ExpansionPanel(
+        headerBuilder: (BuildContext context, bool isExpanded) {
+          return Text(isExpanded ? 'D' : 'C', key: secondPanelKey,);
+        },
+        body: const SizedBox(height: 100.0),
+        canTapOnHeader: true,
+        isExpanded: listState[1],
+      ),
+    ];
+
+    final ExpansionPanelList _expansionListRadio = ExpansionPanelList(
+      expansionCallback: (int _index, bool _isExpanded) {
+        listState[_index] = !listState[_index];
+      },
+      children: _demoItemsRadio,
+    );
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: SingleChildScrollView(
+          child: _expansionListRadio,
+        ),
+      ),
+    );
+
+    // Initializes with all panels closed
+    expect(find.text('A'), findsOneWidget);
+    expect(find.text('B'), findsNothing);
+    expect(find.text('C'), findsOneWidget);
+    expect(find.text('D'), findsNothing);
+
+    await tester.tap(find.byKey(firstPanelKey));
+
+    await tester.pump(const Duration(milliseconds: 200));
+    await tester.pumpAndSettle();
+
+    // The first panel is open
+    expect(listState[0], isTrue);
+
+    await tester.tap(find.byKey(firstPanelKey));
+    await tester.pumpAndSettle();
+
+    // The first panel is closed
+    expect(listState[0], isFalse);
+
+    await tester.tap(find.byKey(secondPanelKey));
+    await tester.pumpAndSettle();
+
+    // The second panel is open
+    expect(listState[1], isTrue);
+
+    await tester.tap(find.byKey(secondPanelKey));
+    await tester.pumpAndSettle();
+
+    // The second panel is closed
+    expect(listState[1], isFalse);
```

Comment:
This does not actually test to see if the panel is open. I would use the `Text` widget values instead, as you did in the previous test.